src/steps/os/macos.rs

```diff
@@ -93,3 +94,79 @@ pub fn run_sparkle(ctx: &ExecutionContext) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn update_xcodes(ctx: &ExecutionContext) -> Result<()> {
+    let xcodes = require("xcodes")?;
+    print_separator("Xcodes");
+
+    let should_ask = !(ctx.config().yes(Step::System)) || (ctx.config().dry_run());
+
+    let releases = ctx.run_type()
+        .execute(&xcodes)
+        .args(["update"])
+        .output_checked_utf8()?.stdout;
```

Comment:
Would you like to give me a output of `xcodes update` so that I can understand the code better?

Reply:
```
1.0 (7B85)
1.5 (7K571)
2.2.1 (8G1165)
2.3 (8M1780)
2.4 (8K1079)
...
15.1 Beta 2 (15C5042i)
15.1 Beta 3 (15C5059c)
15.1 (15C65)
15.2 Beta (15C5500c) (Installed)
15.2 (15C500b)
```

It just updates the list of releases available. As shown above, it also outputs the list at the same time. I can also run this on it's own and run `xcodes list` separately if it's more visibly sensible. I'll put a comment explaining this on the next commit.